src/UDPProtocol.java

```diff
@@ -0,0 +1,83 @@
+import java.io.IOException;
+import java.net.*;
+
+public class UDPProtocol implements Protocol {
+    private DatagramSocket socket;
+    private DatagramPacket sendPacket;
+    private DatagramPacket receivePacket;
```

Comment:
does it need to be a field?

Reply:
Deleted the packet fields and added host address and host port instead.